daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala

```diff
@@ -902,17 +902,13 @@ private[lf] object Speedy {
   private[speedy] final case class KFoldl(
       func: SEValue,
       var list: FrontStack[SValue],
-      frame: Frame,
-      actuals: Actuals,
-      envSize: Int,
   ) extends Kont
       with SomeArrayEquals {
     def execute(acc: SValue, machine: Machine) = {
       list.pop match {
         case None =>
           machine.returnValue = acc
         case Some((item, rest)) =>
-          machine.restoreEnv(frame, actuals, envSize)
```

Comment:
Don't we need this restore in case the function allocates on the stack?

Reply:
Yes, you're right. This will leak stack space otherwise.